src/ripper/ripcddialog.cpp

```diff
@@ -331,3 +335,13 @@ void RipCDDialog::ResetDialog() {
   ui_->yearLineEdit->clear();
   ui_->discLineEdit->clear();
 }
+
+void RipCDDialog::timerEvent(QTimerEvent* e) {
+  if (ripper_->MediaChanged()) {
+    qLog(Debug) << "media changed!";
+    ResetDialog();
+    if (CheckCDIOIsValid()) {
```

Comment:
It looks like it originally just checked that `cdio_` wasn't null. They should have given it a better name here:
https://github.com/clementine-player/Clementine/commit/4251fd43315c176223764ed568e0a77cffae7571

Reply:
That change seems odd to me and there's not rationale for it given in the commit. I would at least now rename it to `EnsureCDIOIsOpened` or, preferrably, move opening of the cdio object back into the class constructor (or at least otherwise treat it internally and not expose this to other classes, ensuring access to resources like this should be encapsulated...)